From: Dan Carpenter <dan.carpenter@linaro.org>
Subject: ceph: fix NULL vs IS_ERR() bug in ceph_zero_partial_page()
Date: Wed, 25 Jun 2025 10:22:11 -0500

The filemap_lock_folio() never returns NULL.  It returns error pointers.
Update the checking to match.

Link: https://lkml.kernel.org/r/685c1424.050a0220.baa8.d6a1@mx.google.com
Signed-off-by: Dan Carpenter <dan.carpenter@linaro.org>
Reviewed-by: Viacheslav Dubeyko <Slava.Dubeyko@ibm.com>
Cc: Christoph Hellwig <hch@lst.de>
Cc: Ilya Dryomov <idryomov@gmail.com>
Cc: Ira Weiny <ira.weiny@intel.com>
Cc: Jens Axboe <axboe@kernel.dk>
Cc: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Cc: Xiubo Li <xiubli@redhat.com>
Cc: Alex Markuze <amarkuze@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/ceph/file.c |   13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

--- a/fs/ceph/file.c~ceph-convert-ceph_zero_partial_page-to-use-a-folio-fix
+++ a/fs/ceph/file.c
@@ -2536,12 +2536,13 @@ static inline void ceph_zero_partial_pag
 	struct folio *folio;
 
 	folio = filemap_lock_folio(inode->i_mapping, offset >> PAGE_SHIFT);
-	if (folio) {
-		folio_wait_writeback(folio);
-		folio_zero_range(folio, offset_in_folio(folio, offset), size);
-		folio_unlock(folio);
-		folio_put(folio);
-	}
+	if (IS_ERR(folio))
+		return;
+
+	folio_wait_writeback(folio);
+	folio_zero_range(folio, offset_in_folio(folio, offset), size);
+	folio_unlock(folio);
+	folio_put(folio);
 }
 
 static void ceph_zero_pagecache_range(struct inode *inode, loff_t offset,
_
